### PR TITLE
REVMI-198 Read lms_user_id from db

### PR DESCRIPTION
**Pull request by crice100**

### CODE REVIEW

File: ecommerce/core/models.py [hunk 1]
```diff
@@ -509,13 +509,40 @@ def access_token(self):
...
+        """
+        # Check the cache for the id
+        cache_key = 'lms_user_id_{user_id}'.format(user_id=self.id)
+        cached_response = DEFAULT_REQUEST_CACHE.get_cached_response(cache_key)
+        if cached_response.is_found:
+            return cached_response.value
+
+        # Calculate the id
+        lms_user_id = self._lms_user_id_from_request()
+        if lms_user_id is not None:
```
Comment (robrap) [inline ecommerce/core/models.py]: If we don't expect it to be updated during the request, we shouldn't do this check and just cache it, even if we are caching/returning None.

File: ecommerce/core/models.py [hunk 2]
```diff
@@ -509,13 +509,40 @@ def access_token(self):
         except Exception:  # pylint: disable=broad-except
             return None
 
-    # TODO Update this to first check the lms_user_id (see REVMI-198). Also consider making this not a property and
-    # renaming it.
-    @property
     def lms_user_id_from_request(self):
```
Comment (robrap) [inline ecommerce/core/models.py:507]: This no longer feels like an appropriate name.

File: ecommerce/core/models.py (hunk 1 above)
Comment (crice100) [reply ecommerce/core/models.py]: If we do that, I was concerned that in the future we might add new middleware that runs very early in the request and sees the id is None. Then our existing TrackingMiddleware (which runs later) updates the id, but the new id won't be used until subsequent requests due to the caching. 

I suppose we could catch this situation with logging, but the current approach seemed safer. Thoughts?

File: ecommerce/core/models.py (hunk 2 above)
Comment (crice100) [reply ecommerce/core/models.py:507]: Maybe _get_cached_lms_user_id()_?

File: ecommerce/core/models.py (hunk 1 above)
Comment (robrap) [reply ecommerce/core/models.py]: Right.  Should all this code be in the middleware?

File: ecommerce/core/models.py (hunk 2 above)
Comment (robrap) [reply ecommerce/core/models.py:507]: `get_and_cache_lms_user_id()`?

**Pull request merged**